src/Regulator.h

```diff
@@ -176,48 +231,57 @@ class Regulator : public RingBuffer
     virtual bool getStats(IOStat* stat, bool reset);
 
    private:
-    void shimFPP(const int8_t* buf, int seq_num);
     void pushPacket(const int8_t* buf, int seq_num);
     void updatePushStats(int seq_num);
     void pullPacket();
     void updateTolerance();
     void setFPPratio(int len);
     void processPacket(bool glitch);
-    void processChannel(int ch, bool glitch, int packetCnt, bool lastWasGlitch);
-
+    void burg(bool glitch);
+    sample_t bitsToSample(int ch, int frame);
+    void sampleToBits(sample_t sample, int ch, int frame);
+    void floatBufToXfrBuffer();
+    void xfrBufferToFloatBuf();
+    void toFloatBuf(qint16* in);
+    void fromFloatBuf(qint16* out);
+    void zeroTmpFloatBuf();
+    int mPacketsInThePast;
     bool mInitialized;
     int mNumChannels;
     int mAudioBitRes;
-    int mFPP;
+    int mLocalFPP;
     int mPeerFPP;
     int mSampleRate;
+    int mPcnt;
+    std::vector<float> mTmpFloatBuf;
+    std::vector<Channel*> mChanData;
+    BurgAlgorithm* ba;
```

Comment:
any raw pointers should be initialized to NULL either in header or constructor, to avoid delete's of initialized memory (null is ok)

Reply:
sprinkled = nullptr througout the .h